pkg/client/cache/thread_safe_store.go

```diff
@@ -179,6 +179,9 @@ func (c *threadSafeMap) ByIndex(indexName, indexKey string) ([]interface{}, erro
 }
 
 func (c *threadSafeMap) ListIndexFuncValues(indexName string) []string {
```

Comment:
I don't see a test for this function, either. I'm not 100% sure this function makes any sense at all. but at least this seems like an improvement.